src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -65,6 +63,11 @@ public void BeginRequest(HttpContext httpContext, HostingApplication.Context con
 
             if (context.Activity is Activity activity)
             {
+                // Add some ASP.NET Core specific properties as tags
+                activity.AddTag("aspnet.traceidentifier", httpContext.TraceIdentifier);
```

Comment:
This is really only useful in the IIS and HTTP.sys. They flow an ID that is can be used to correlate with their lower level ETW traces.

Reply:
Do you think we could pick a name that is more IIS/HTTP.sys specific and only include the identifier when asp.net is run in that hosting environment?

Reply:
Is the ASP.NET prefix not sufficient for that? Kestrel invents its own ID today as well. There's no easy way from here to tell whether we're running in IIS or HTTP.sys. Since the activity is created later then the server can detect, it can't do anything by default since it doesn't have that reference. 

Another possibility is we don't add anything, and developers can write code to enrich the activity with the trace identifier when they need this functionality. Not ideal but the only places we get access to the http context and the activity is middleware or our `OnActivityImport` diagnostic source event.

Reply:
I've removed it.